Add case studies link at the top of the home page

visortelle opened the pull request:
The "Case Studies" section is currently placed at the bottom of the home page.

I think it's good for the project to show that many companies use Pulsar at the very beginning of the page.

This PR adds this block to the first homepage screen.

Live demo: https://pulsar-site-2024-updates-git-case-studies-at-top-visortelle.vercel.app/

<img width="1440" alt="Screenshot 2024-03-01 at 2 58 42 PM" src="https://github.com/apache/pulsar-site/assets/9302460/b719405d-06d6-45ed-9e61-af87061ccd90">

I'm not sure if is it allowed to mention specific companies. We can somehow rephrase the text if not.

- Apache Pinot has such a block that mentions LinkedIn on the homepage.

<img width="777" alt="Screenshot 2024-03-01 at 2 59 53 PM" src="https://github.com/apache/pulsar-site/assets/9302460/636f9abd-6ac9-4e8b-8e83-566d20355187">

- Apache Kafka has it at the top of its homepage.

<img width="828" alt="Screenshot 2024-03-01 at 3 00 56 PM" src="https://github.com/apache/pulsar-site/assets/9302460/c003af34-c149-46db-9bdd-8d3f70b84f53">

cc: @dave2wave @lhotari

dave2wave commented:
Please don't mention company names on the one page. We can have a secondary page, but that has to make it so that we update and display the list in a fair and unbiased manner - sorted by name and a clear way for PRs to be submitted

dave2wave commented:
Pinot is not doing this properly.

visortelle commented:
@dave2wave is it better now?

<img width="1440" alt="Screenshot 2024-03-01 at 4 12 56 PM" src="https://github.com/apache/pulsar-site/assets/9302460/ee777a21-65b7-42ad-b006-9e2d62081245">

The link refers to https://pulsar.apache.org/case-studies/

The list of companies on this page is alphabetically sorted.

The place where to find and update the page data is mentioned here:

https://pulsar.apache.org/contribute/site-intro/#how-to-update-data-driven-pages

<img width="775" alt="Screenshot 2024-03-01 at 4 16 21 PM" src="https://github.com/apache/pulsar-site/assets/9302460/0b84fe8c-eb14-45d3-811d-5c84dafcf358">

visortelle commented:
It’s probably too much text now for the first screen. However, in my opinion, having a link to case studies on the first screen is more important, than repeating the same phrase about messaging and streaming two times.

cc @asafm

visortelle commented:
@dave2wave 
> Pinot is not doing this properly.

I like the Pinot project, but I would expect the "not right" section to also be fixed in the near future.

dave2wave commented:
I would consider adding a sentence and link on data driven pages that points to how to contribute updates. Something like "Are we missing any <X>? Please contribute updates by following the instructions [here](link to page)

Otherwise great!

visortelle commented:
@dave2wave I added the link you've asked about.

Live preview: https://pulsar-site-2024-updates-git-case-studies-at-top-visortelle.vercel.app/case-studies

<img width="1440" alt="Screenshot 2024-03-02 at 12 05 44 PM" src="https://github.com/apache/pulsar-site/assets/9302460/3bdac4f4-a793-4ff9-8258-d460aed66edd">
<img width="1440" alt="Screenshot 2024-03-02 at 12 06 04 PM" src="https://github.com/apache/pulsar-site/assets/9302460/62e04ba6-85ca-4efc-bfbb-7229eb28c5cb">
<img width="1440" alt="Screenshot 2024-03-02 at 12 06 20 PM" src="https://github.com/apache/pulsar-site/assets/9302460/a03fd349-4ee2-4884-954c-f74a20ccfd7f">

Also, I updated the "How to update data-driven pages?" section a bit. I added the direct "edit file content" link on GitHub for most pages.

<img width="861" alt="Screenshot 2024-03-02 at 12 07 06 PM" src="https://github.com/apache/pulsar-site/assets/9302460/6352eef7-da10-452a-8f61-27f4bdbbb29d">

dave2wave commented:
Excellent! Thank you!

asafm commented:
@visortelle We should stop merging stuff without review.

Notes from me:
1. The box you added looks "stuck" there - do not fit design-wise. I wouldn't add it there.
2. This information should fit in the big PR, which is now entering the final phase and adds the case studies section. I think it makes more sense there.
3. The design of "Are we missing anything?" doesn't look good. Let's remove it and get some design advice from Emidio.

visortelle commented:
@asafm Regarding review. I'm asking for opinions if it's not some cosmetic styling changes.

@dave2wave liked it. I don't see how it much differs from the formal review, but ok I'll wait for someone's approval next time.

**1**, **2**. I checked the Figma design and the PR you're talking about and didn't find the case studies section you're talking about.

> The box you added looks "stuck" there - do not fit design-wise. I wouldn't add it there.
This information should fit in the big PR, which is now entering the final phase and adds the case studies section. I think it makes more sense there.

**Figma**
<img width="591" alt="Screenshot 2024-03-04 at 7 59 02 PM" src="https://github.com/apache/pulsar-site/assets/9302460/9a6d3094-43dd-4ad2-b204-c8aafe84d2aa">

**The PR**
<img width="1433" alt="Screenshot 2024-03-04 at 7 59 45 PM" src="https://github.com/apache/pulsar-site/assets/9302460/bd34bde1-5aa4-4152-a88a-6b9f01d22d4c">

We already have this section at the bottom of the page:

**Current pulsar.apache.org**
<img width="1440" alt="Screenshot 2024-03-04 at 8 01 05 PM" src="https://github.com/apache/pulsar-site/assets/9302460/64caf488-7486-4b2c-9a63-33ba6a02d6a5">

**3.** 

> The design of "Are we missing anything?" doesn't look good. Let's remove it and get some design advice from Emidio.

Why remove? Let's just wait Emidio will improve it and I'll apply the new changes.

asafm commented:
Sorry, I was confused.
Today, we have the Pulsar Users section with the sentence beneath it, right?
<img width="737" alt="image" src="https://github.com/apache/pulsar-site/assets/989425/6553743f-d5f3-4045-b02a-5ac042daaf0d">

We can rework that sentence to be more "inviting". 
The Pulsar website was designed around the notion of gradual time to understand and target primarily people who haven't heard of it before.

When you hit Pulsar main page for the first time, you read the first sentence to get what it is all about.
<img width="586" alt="image" src="https://github.com/apache/pulsar-site/assets/989425/7e8beca7-8279-4615-af9d-ae01ac219566">

It took 10 seconds. You are intrigued. You scroll down.
You spent 1 more minute and read "What is Pulsar".
You want to learn more. Yo scroll down.
You spend 2 minutes reading the top features list.
You spend 3 minutes reading "How does pulsar work" section.
Now you get to Pulsar Users, and you read:

<img width="781" alt="image" src="https://github.com/apache/pulsar-site/assets/989425/55b89785-5ffc-41fd-b516-4dc2da27db4f">

That gives you the trust you need.
You read on, and you gain more trust by reading about the community.

The alternative:
When you hit Pulsar main page for the first time, you read the first sentence to get what it is all about.
Then it suddenly says it is used by many companies and you are recommended to read the case studies.

To me personally it feels a bit awkward.

Maybe if were to place a single sentence, with really nice design, with no links to case studies (yet - it will be at Pulsar Users section) - it can work better.

visortelle commented:
> You spend 3 minutes reading "How does pulsar work" section.
> Now you get to Pulsar Users, and you read:

@asafm It doesn't work this way. If you want, I can search for some research on this topic.

visortelle commented:
If even top SEO/audience analytics companies do that...

![Screenshot 2024-03-05 at 12 39 42 AM](https://github.com/apache/pulsar-site/assets/9302460/b6c9e406-a05d-4d44-9539-d5d271e0dfb9)
![Screenshot 2024-03-05 at 12 40 04 AM](https://github.com/apache/pulsar-site/assets/9302460/9f8a07d1-01ab-4905-91e0-bfadfb1e7332)
![Screenshot 2024-03-05 at 12 40 21 AM](https://github.com/apache/pulsar-site/assets/9302460/51d51f14-3b71-4891-9731-5bbcfd1181df)
